### PR TITLE
BT 33842

### PR DESCRIPTION
**Pull request by schmitz-ilias**
This PR fixes [33842](https://mantis.ilias.de/view.php?id=33842). Would you do the honors again @alex40724 ?

### CODE REVIEW

File: Services/Search/classes/Lucene/class.ilLuceneSearchObjectListGUIFactory.php [hunk 1]
```diff
@@ -58,7 +53,7 @@ public static function factory(string $a_type) : ilObjectListGUI
         $full_class = "ilObj" . $class . "ListGUI";
 
         include_once($location . "/class." . $full_class . ".php");
-        $item_list_gui = new $full_class();
+        $item_list_gui = new $full_class(ilObjectListGUI::CONTEXT_SEARCH);
```
Comment (alex40724) [inline Services/Search/classes/Lucene/class.ilLuceneSearchObjectListGUIFactory.php:56]: I just did a quick check, if this parameter is correctly passed by the overwritten constructors. This is not always the case, I found: BookingPool, Forum, ItemGroup, LearningSequence, OrgUnit, RootFolder, SAHSLearningModule, Session, StudyProgramme, Survey, RemoteObjectBase, example:

```
class ilObjBookingPoolListGUI extends ilObjectListGUI
{
    protected \ILIAS\BookingManager\StandardGUIRequest $book_request;

    public function __construct()
    {
        global $DIC;

        $this->ctrl = $DIC->ctrl();
        $this->lng = $DIC->language();
        parent::__construct();
        $this->book_request = $DIC->bookingManager()
                                  ->internal()
                                  ->gui()
                                  ->standardRequest();
    }
...
}
```

Actually I am not sure, what CONTEXT_SEARCH controls exactly after quickly looking at the code.

But all these constructors might need to be fixed, or am I wrong? If yes, we might merge this one and create a separate PR.

**Review by alex40724 — CHANGES_REQUESTED** [review]

File: Services/Search/classes/Lucene/class.ilLuceneSearchObjectListGUIFactory.php (hunk 1 above)
Comment (schmitz-ilias) [reply Services/Search/classes/Lucene/class.ilLuceneSearchObjectListGUIFactory.php:56]: Setting the context to CONTEXT_SEARCH is needed here to make sure that uploading files to container objects that show up in the search results is disabled (the file dropzones are causing an error, and this functionality shouldn't be enabled in the search results anyways).

As far as I can see you are right, all overwritten constructors that don't already pass the context should be fixed to do so.

**Review by alex40724 — COMMENTED** [review]
@smeyer-ilias Could you take care of this one, please.

**Review by alex40724 — COMMENTED** [review]
.

**Review by alex40724 — APPROVED** [review]

**Comment by smeyer-ilias** [comment]
Thanks to both of you. 
I will create a PR which adds the context-type to constructors

**Pull request merged**